node/silkworm/etl/collector.hpp

```diff
@@ -76,7 +76,7 @@ class Collector {
 
     //! \brief Returns the hex representation of current load key (for progress tracking)
     [[nodiscard]] std::string get_load_key() const {
-        std::unique_lock l{mutex_};
```

Comment:
This is not correct. Mutexes  [do synchronize memory](https://stackoverflow.com/questions/15313296/release-acquire-semantics-wrt-stdmutex). [atomic_thread_fence](https://en.cppreference.com/w/cpp/atomic/atomic_thread_fence) is useful for atomic variables, which `loading_key_` is not. The previous code with mutex was fine.

Reply:
> The previous code with mutex was fine.

Well actually it wasn't : see https://erigon5900.weblogix.it/er.log
![image](https://user-images.githubusercontent.com/15280851/155116070-aa495357-4cfb-46c7-b3f6-fc45497dcb55.png)
![image](https://user-images.githubusercontent.com/15280851/155116185-d1f876c1-f015-43bc-afdf-03262a8c4514.png)
![image](https://user-images.githubusercontent.com/15280851/155116260-55b2edaa-81c0-4b06-9e32-39805cf933b0.png)

Reply:
atomic_thread_fence is useful for NON atomic variables (like string)

Reply:
Oh well missed that.